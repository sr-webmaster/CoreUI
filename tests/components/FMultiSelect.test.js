import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import { clear } from 'jest-date-mock'
import map from 'lodash/map'
import Component from '@freshinup/core-ui/src/components/FMultiSelect.vue'

const FIXTURE_STATUSES = [
  { id: 1, name: 'Draft' },
  { id: 2, name: 'Pending' },
  { id: 3, name: 'Confirmed' },
  { id: 4, name: 'Past' },
  { id: 5, name: 'Cancelled' }
]

const allSelected = FIXTURE_STATUSES.map(item => item.id)

describe('FMultiSelect', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          value: [],
          items: FIXTURE_STATUSES
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
      clear()
    })

    test('toggle function select all', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: [],
          items: FIXTURE_STATUSES
        }
      })
      wrapper.vm.toggle()
      await wrapper.vm.$nextTick()
      expect(wrapper.emitted()['input']).toBeTruthy()
      expect(wrapper.emitted()['input'][0][0]).toEqual(map(FIXTURE_STATUSES, 'id'))
    })

    test('toggle function clear all', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: allSelected,
          items: FIXTURE_STATUSES
        }
      })
      wrapper.vm.toggle()
      await wrapper.vm.$nextTick()
      expect(wrapper.emitted()['input']).toBeTruthy()
      expect(wrapper.emitted()['input'][0][0]).toEqual([])
    })
  })

  describe('Computed', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('selectAll', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: allSelected,
          items: FIXTURE_STATUSES
        }
      })
      expect(wrapper.vm.selectAll).toBeTruthy()
      expect(wrapper.vm.selectSome).toBeFalsy()
    })
    test('selectSome', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: [ 1, 2 ],
          items: FIXTURE_STATUSES
        }
      })
      expect(wrapper.vm.selectAll).toBeFalsy()
      expect(wrapper.vm.selectSome).toBeTruthy()
    })
    test('icon for no select', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: [],
          items: FIXTURE_STATUSES
        }
      })
      expect(wrapper.vm.icon).toBe('far fa-square')
    })
    test('icon for some select', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: [ 1, 2 ],
          items: FIXTURE_STATUSES
        }
      })
      expect(wrapper.vm.icon).toBe('fa-minus-square')
    })
    test('icon for select all', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        propsData: {
          value: allSelected,
          items: FIXTURE_STATUSES
        }
      })
      expect(wrapper.vm.icon).toBe('fa-check-square')
    })
  })
})
