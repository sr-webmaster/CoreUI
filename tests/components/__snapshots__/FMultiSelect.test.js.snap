// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FMultiSelect Snapshots defaults 1`] = `
<div
  class="v-input v-text-field v-select theme--light"
  role="combobox"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-select__slot"
      >
        <div
          class="v-select__selections"
        >
          <input
            aria-readonly="false"
            autocomplete="on"
            readonly="readonly"
            type="text"
          />
        </div>
        <div
          class="v-input__append-inner"
        >
          <div
            class="v-input__icon v-input__icon--append"
          >
            <i
              aria-hidden="true"
              class="v-icon material-icons theme--light"
            >
              arrow_drop_down
            </i>
          </div>
        </div>
      </div>
      <div
        class="v-menu"
      >
        <div
          class="v-menu__content theme--light "
          style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
        >
          <div
            class="v-select-list v-card theme--light"
          >
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__action"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon far fa-square theme--light"
                    />
                  </div>
                   
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      All Options
                    </div>
                  </div>
                </a>
              </div>
               
              <hr
                class="mt-2 v-divider theme--light"
              />
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__action"
                  >
                    <div
                      class="v-input v-input--selection-controls v-input--checkbox theme--light"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-input--selection-controls__input"
                          >
                            <input
                              aria-checked="false"
                              role="checkbox"
                              type="checkbox"
                              value=""
                            />
                            <div
                              class="v-input--selection-controls__ripple"
                            />
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              check_box_outline_blank
                            </i>
                          </div>
                        </div>
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      [object Object]
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__action"
                  >
                    <div
                      class="v-input v-input--selection-controls v-input--checkbox theme--light"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-input--selection-controls__input"
                          >
                            <input
                              aria-checked="false"
                              role="checkbox"
                              type="checkbox"
                              value=""
                            />
                            <div
                              class="v-input--selection-controls__ripple"
                            />
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              check_box_outline_blank
                            </i>
                          </div>
                        </div>
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      [object Object]
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__action"
                  >
                    <div
                      class="v-input v-input--selection-controls v-input--checkbox theme--light"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-input--selection-controls__input"
                          >
                            <input
                              aria-checked="false"
                              role="checkbox"
                              type="checkbox"
                              value=""
                            />
                            <div
                              class="v-input--selection-controls__ripple"
                            />
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              check_box_outline_blank
                            </i>
                          </div>
                        </div>
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      [object Object]
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__action"
                  >
                    <div
                      class="v-input v-input--selection-controls v-input--checkbox theme--light"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-input--selection-controls__input"
                          >
                            <input
                              aria-checked="false"
                              role="checkbox"
                              type="checkbox"
                              value=""
                            />
                            <div
                              class="v-input--selection-controls__ripple"
                            />
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              check_box_outline_blank
                            </i>
                          </div>
                        </div>
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      [object Object]
                    </div>
                  </div>
                </a>
              </div>
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__action"
                  >
                    <div
                      class="v-input v-input--selection-controls v-input--checkbox theme--light"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-input--selection-controls__input"
                          >
                            <input
                              aria-checked="false"
                              role="checkbox"
                              type="checkbox"
                              value=""
                            />
                            <div
                              class="v-input--selection-controls__ripple"
                            />
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              check_box_outline_blank
                            </i>
                          </div>
                        </div>
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-list__tile__content"
                  >
                    <div
                      class="v-list__tile__title"
                    >
                      [object Object]
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="v-text-field__details"
    >
      <div
        class="v-messages theme--light"
      >
        <div
          class="v-messages__wrapper"
        />
      </div>
    </div>
  </div>
</div>
`;
